device/alpaka/src/utils/utils.hpp

```diff
@@ -8,34 +8,47 @@
 #pragma once
 
 #include <alpaka/alpaka.hpp>
-#include <alpaka/example/ExampleDefaultAcc.hpp>
-
-#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
-#include <vecmem/utils/cuda/copy.hpp>
-#endif
-
-#ifdef ALPAKA_ACC_GPU_HIP_ENABLED
-#include <vecmem/utils/hip/copy.hpp>
-#endif
-
-#include <vecmem/utils/copy.hpp>
 
 namespace traccc::alpaka {
 
 using Dim = ::alpaka::DimInt<1>;
 using Idx = uint32_t;
 using WorkDiv = ::alpaka::WorkDivMembers<Dim, Idx>;
 
-using Acc = ::alpaka::ExampleDefaultAcc<Dim, Idx>;
+// Get alpaka accelerator - based on alpaka/examples/ExampleDefaultAcc.hpp
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+using Acc = ::alpaka::AccGpuCudaRt<Dim, Idx>;
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+using Acc = ::alpaka::AccGpuHipRt<Dim, Idx>;
+#elif defined(ALPAKA_ACC_SYCL_ENABLED)
+#if defined(ALPAKA_SYCL_ONEAPI_CPU)
+using Acc = ::alpaka::AccCpuSycl<Dim, Idx>;
+#elif defined(ALPAKA_SYCL_ONEAPI_FPGA)
+using Acc = ::alpaka::AccFpgaSyclIntel<Dim, Idx>;
+#elif defined(ALPAKA_SYCL_ONEAPI_GPU)
+using Acc = ::alpaka::AccGpuSyclIntel<Dim, Idx>;
+#endif
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED)
+using Acc = ::alpaka::AccCpuThreads<Dim, Idx>;
+#else
+#error "No supported backend selected." //we definitely want to fail the build if no matching accelerator is found
+#endif
```

Comment:
Could this not appear just once? In this PR this is the second time I see the exact same thing. Or is it not exactly the same?

Reply:
I think one of them defines Accelerator _tagS_ and this one is just Accelerators; I noticed the same but I think they are different.

Reply:
Right, the tags are a way to refer to different accelerators without having to include `alpaka/alpaka.hpp` at which point you have to use the device compiler. It's annoying to have basically the same logic twice but it lets us define`Acc` in the source and then `Tag` in the header.